Libraries/LibWeb/Loader/ResourceLoader.cpp

```diff
@@ -320,7 +322,9 @@ void ResourceLoader::load(LoadRequest& request, GC::Root<SuccessCallback> succes
 
         // When resource URI is a directory use file directory loader to generate response
         if (resource.value()->is_directory()) {
-            respond_directory_page(request, resource.value()->file_url(), success_callback, error_callback);
+            auto url = URL::Parser::basic_parse(resource.value()->file_url());
+            VERIFY(url.has_value());
```

Comment:
Doing this a bunch makes me wonder if we should just have an explicit infallible URL parser

Reply:
Yeah, though each of those cases did take a fair amount of auditing however to figure out if I thought there were okay or not to assume parsing succeeded or not. I suppose this is the type of case that MUST type macro would be useful.